src/v/cluster/metadata_cache.cc

```diff
@@ -248,4 +276,47 @@ metadata_cache::get_default_shadow_indexing_mode() const {
     }
     return m;
 }
+
+ss::future<> metadata_cache::broadcast_disk_space_state(
+  const storage::disk_space_alert state) {
+    return container().invoke_on_others(
+      [state](metadata_cache& mc) { mc.set_disk_space_state(state); });
+}
+
+void metadata_cache::set_disk_space_state(const storage::disk_space_alert s) {
+    _disk_space_state = s;
+    vlog(clusterlog.debug, "set_disk_space_state {}", s);
+}
+
+bool metadata_cache::should_reject_writes() const {
+    return _disk_space_state == storage::disk_space_alert::degraded;
+}
+
+void metadata_cache::maybe_update_cluster_health() {
+    if (_refresh_gate.is_closed()) {
+        return;
+    }
+    ssx::spawn_with_gate(_refresh_gate, [this]() -> ss::future<> {
+        // This function ensures that this node's cluster health data is not too
+        // stale (i.e. <= 5 seconds). We actually get the updates via the
+        // `set_disk_space_state()` callback from health monitor. This way, we
+        // don't have to allocate, enumerate all nodes, and destroy a cluster
+        // health report just to get the current drive health enum value.
+        co_await _health_monitor.local()
+          .maybe_refresh_cluster_health()
```

Comment:
would it be possible to call here `health_monitor_frontend::get_cluster_health` ?. This way the cache wouldn't have to trigger `health_monitor` refresh (which is its internal logic) health monitor would decide if it is required to update the cache based on its internal logic.

Reply:
This way we wouldn't have to deal with special shards as each shard would have its own timer that would communicate with its own shard local `health_moniotr_frontend`

Reply:
I think you are proposing making cross-core calls to health monitor backend on every produce request. How expensive is that?

Reply:
Also note that the produce handler is synchronous (non-futurized)

Reply:
i am not really proposing x-core requests here, i was thinking about the timer in `metadata_cache` querying health monitor for `get_cluster_health` then extract the disk_alert and cache it on every core.

Reply:
Ah, yeah, I misread this. Was just going to edit my comment. ;-) This is slightly less efficient but much nicer layering as you note. Are you cool with doing this on a specific (metadata_cache) refresh shard and broadcasting to the others (assuming I fix the awkward start() semantics)? Or, would you rather each metadata_cache shard just do it separately. (I think I prefer the former).

Reply:
Next update will keep the single-shard poll and broadcast behavior of metadata_cache. We can continue the discussion there and see what you prefer.

Reply:
See latest patch with your suggestion implemented. Please resolve this `outdated` thread and continue on the new patch.